quaddtype/meson.build

```diff
@@ -1,20 +1,72 @@
-project('quaddtype', 'c', 'cpp', default_options : ['cpp_std=c++17', 'b_pie=true'])
+project('quaddtype', 'c', 'cpp', default_options : ['cpp_std=c++20', 'b_pie=true'])
 
 py_mod = import('python')
 py = py_mod.find_installation()
 
 c = meson.get_compiler('c')
 
-sleef_dep = c.find_library('sleef')
-sleefquad_dep = c.find_library('sleefquad')
+is_windows = build_machine.system() == 'windows'
 
-incdir_numpy = run_command(py,
-  [
-    '-c',
-    'import numpy; import os; print(os.path.relpath(numpy.get_include()))'
-  ],
-  check: true
-).stdout().strip()
+incdir_numpy = ''
+if is_windows
+  add_project_arguments('-DWIN32', '-D_WINDOWS', language : ['c', 'cpp'])
+  
+  conda_env_dir = run_command('cmd', '/c', 'echo %CONDA_PREFIX%', check: true).stdout().strip()
+  sleef_include_dir = conda_env_dir + '\\Library\\include'
+  sleef_library_dir = conda_env_dir + '\\Library\\lib'
+  incdir_numpy = run_command('cmd', '/c', 'echo %NUMPY_INCLUDE_DIR%', check: true).stdout().strip()
+  
+  add_project_arguments('-I' + sleef_include_dir, language: ['c', 'cpp'])
+  add_project_link_arguments('-L' + sleef_library_dir, language: ['c', 'cpp'])
+  
+  sleef_lib = c.find_library('sleef', dirs: [sleef_library_dir])
+  sleefquad_lib = c.find_library('sleefquad', dirs: [sleef_library_dir])
+  
+  sleef_dep = declare_dependency(include_directories: include_directories(sleef_include_dir),
+                                 dependencies: [sleef_lib, sleefquad_lib])
+else
+  # Linux and macOS configuration
+
+  # Get the SLEEF path
+  sleef_path = run_command('bash', '-c', 'echo $SLEEF_PATH', check: false).stdout().strip()
+  if sleef_path == ''
+    sleef_path = run_command('bash', '-c', 'echo $CONDA_PREFIX', check: false).stdout().strip()
+  endif
+
+  if sleef_path == ''
+    error('SLEEF_PATH or CONDA_PREFIX environment variable is not set')
+  endif
+
+  add_project_link_arguments('-L' + sleef_path + '/lib', language: ['c', 'cpp'])
+
+  sleef_include_dir = sleef_path + '/include'
+  sleef_library_dir = sleef_path + '/lib'
+
+  add_project_arguments('-I' + sleef_include_dir, language: ['c', 'cpp'])
+  add_project_link_arguments('-L' + sleef_library_dir, language: ['c', 'cpp'])
+  
+  sleef_dep = c.find_library('sleef', dirs: [sleef_library_dir])
+  sleefquad_dep = c.find_library('sleefquad', dirs: [sleef_library_dir])
+endif
```

Comment:
I'm pretty sure the correct way to do this is to use `find_library` as you had it before. Why do you need all of this OS-specific stuff? Is it to find the conda package?

I think NumPy uses pkgconfig to avoid needing to spell out exactly how libraries were installed to build against them. You can use pkgconfig on windows in CI by installing pkgconfiglite. I've never actually needed to figure this out though so unfortunately can't help much. Ralf or someone else at Quansight probably knows a lot more about the "correct" way to handle third party deps in a meson.build on windows.